### DISCUSSION TITLE
Do canceled deployments count toward deployment quota?

### DISCUSSION THREAD

**Discussion by lukewestby** [vercel/vercel#5716] [Help] [answered]
Our repo receives commits from an outside source via the GitHub API from time to time. Changes in the relevant directories don't require the site to be rebuilt so we are using a custom ignore build command to skip those changes. The command works, but it's not clear to me if deployments started by those changes incur a deduction on our daily deployment quota.

**Comment by RizBizKits** [comment, marked as answer]
Hey! Currently: **yes**. Canceled deployments via the `Ignored Build Step` are counted. Let us know if ya'll have any feedback/use-cases around this that we can prioritize 🙌 

_p.s. I've also replied back to your tweet [here](https://twitter.com/rizbizkits/status/1533053209115254784)._

**Reply by ChristianIvicevic** [reply]
Hey there, allow me to hijack this thread to share my use-case.

I am currently running the free Hobby tier of Vercel and thus have the daily quota of 100 deployments. Earlier today I have enabled Renovate to keep my packages up to date which causes a lot of noise and preview deployments to happen. Suppose it initially creates 15 PRs for updates in my monorepo and thus 15 preview deployments. When I merge one of those PRs, the other 14 PRs left get rebased and thus trigger another set of 14 preview deployments in addition to the production deployment. Similarly whenever I push the `main` branch, Renovate will pick it up to rebase the PRs and run the GitHub actions with my workflows eating through all of my deployments really fast.

Since I don't need those renovate update branches to be deployed as previews I was considering to introduce a `Ignored Build Step` command to only deploy the `main` branch for now, and maybe `staging` later down the road. Hearing that those ignored builds still count for the quota my PRs display that not all checks pass, since Vercel reports the deployment failed.

As a short-term solution I might consider removing the Vercel integration and deploying from within the GitHub workflow conditionally using a CLI. But setting this up is a bit more complex in comparison to the convenience of the automated integration.

Anyways, I hope my use-case is a good example for when the `Ignored Build Step` should not really count against the quota. Maybe in the long run it would be convenient in the Vercel Dashboard to define a custom blacklist for branch names such as `renovate/*` to skip deployment of those altogether.
